iothub/device/src/Exceptions/IotHubClientErrorCode.cs

```diff
@@ -125,5 +125,10 @@ public enum IotHubClientErrorCode
         /// guide for more information.
         /// </remarks>
         Unauthorized,
+
+        /// <summary>
+        /// The request failed because of TLS authentication error.
```

Comment:
Throwing a question out there - what would be our recommendation for this case? Check your device certificates to ensure they are up-to-date?